src/conduits/RwaMultiSwapOutputConduit.sol

```diff
@@ -0,0 +1,462 @@
+        _;
+    }
+
+    /**
+     * @notice Defines addresses and gives `msg.sender` admin access.
+     * @param _dai DAI contract address.
+     */
+    constructor(address _dai) public {
+        require(_dai != address(0), "RwaMultiSwapOutputConduit/wrong-dai-address");
+
+        dai = DaiAbstract(_dai);
+
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    /*//////////////////////////////////
+               Authorization
+    //////////////////////////////////*/
+
+    /**
+     * @notice Grants `usr` admin access to this contract.
+     * @param usr The user address.
+     */
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+
+    /**
+     * @notice Revokes `usr` admin access from this contract.
+     * @param usr The user address.
+     */
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+
+    /**
+     * @notice Grants `usr` operator access to this contract.
+     * @param usr The user address.
+     */
+    function hope(address usr) external auth {
+        can[usr] = 1;
+        emit Hope(usr);
+    }
+
+    /**
+     * @notice Revokes `usr` operator access from this contract.
+     * @param usr The user address.
+     */
+    function nope(address usr) external auth {
+        can[usr] = 0;
+        emit Nope(usr);
+    }
+
+    /**
+     * @notice Grants `usr` push access to this contract.
+     * @param usr The user address.
+     */
+    function mate(address usr) external auth {
+        may[usr] = 1;
+        emit Mate(usr);
+    }
+
+    /**
+     * @notice Revokes `usr` push access from this contract.
+     * @param usr The user address.
+     */
+    function hate(address usr) external auth {
+        may[usr] = 0;
+        emit Hate(usr);
+    }
+
+    /**
+     * @notice Whitelist `who` address for `pick`.
+     * @param who The user address.
+     */
+    function kiss(address who) external auth {
+        bud[who] = 1;
+        emit Kiss(who);
+    }
+
+    /**
+     * @notice Remove `who` address from `pick` whitelist.
+     * @param who The user address.
+     */
+    function diss(address who) external auth {
+        if (to == who) to = address(0);
+        bud[who] = 0;
+        emit Diss(who);
+    }
+
+    /**
+     * @notice Whitelist `psm` address for `hook`.
+     * @param _psm The PSM address.
+     */
+    function clap(address _psm) external auth {
+        require(PsmAbstract(_psm).dai() == address(dai), "RwaMultiSwapOutputConduit/wrong-dai-for-psm");
+
+        // Check if GEM decimals is not greater then DAI decimals. We assume that DAI will alway have 18 decimals
+        // Does decimals should be less then 18, according to AuthGemJoin5 check (decimals < 18)
+        require(
+            GemAbstract(GemJoinAbstract(PsmAbstract(_psm).gemJoin()).gem()).decimals() <= 18,
+            "RwaMultiSwapOutputConduit/gem-decimals-overflow"
+        );
```

Comment:
Where exactly? I'm looking at this.

https://github.com/makerdao/rwa-toolkit/blob/92c79aac24ef7645902ce4be57ba41b19e6c7dd5/src/conduits/RwaSwapOutputConduit.sol#L159-L160

Reply:
Ahh, true

Reply:
In case we have a different join which support more then 18 decimals then we will be able to add psm with unsupported decimals

Reply:
All PSMs use `AuthGemJoin`... I don't think we need to worry about this.
If we remove the check and stick to safe-math, we're fine, as it would simply revert on `_sub`.